### DISCUSSION TITLE
Z2M fa il pairing solo dal coordinator

### DISCUSSION THREAD

**Discussion by agnus777** [indomus/forum#3044] [ZigBee, Z-Wave, Thread, Matter]
Buonasera,

ho provato a chiedere nella chat telegram ma non sono riuscito ad avere risposta, vi spiego il problema:

Su HA sono passato da ZHA a Z2M perchè tutti dicevano che il secondo funziona meglio e l'integrazione dell'ultimo sensore per la tenda in ZHA mi dava problemi, così ho preso, attivato Z2M e rifatto il pairing di tutti i device, tutto bene fino a che , di punto in bianco, sono spariti i files dalla cartella config. tutto eccetto la cartella zigbeemqtt. Quindi ho preso un backup completo e l'ho ripristinato, aggiornato il sistema e ripartito con il pairing, i primi 7-8 devices sono stati riconosciuti perfettamente, poi ha smesso di vedere tutte le periferiche che non fossero nel raggio diretto del coordinator.
tutt'ora riesco a fare i pairing solo vicino all'antenna, una volta effettuato il pairing e spostati vicini ad altri router, funzionano normalmente.

domanda:
c'è qualcuno che sa cosa possa essere successo?
esiste un modo per ripristinare la funzione di accoppiamento da qualunque punto coperto con la rete?

grazie in anticipo

**Comment by trebestie** [comment]
Mi era successo una mezza vita fa.dopo avere pasticciato con un aggiornamento.
Installazione pulita di z2m, resettato tutto, fatto il pairing prima dei router e poi degli end device nella loro posizione finale.
Se non hai dei dispositivi touchlink non è nemmeno un lavoro lungo.

**Reply by KC-inDomus** [reply]
(se posso: effettuare il pairing da router anziché da coordinator non dà alcun beneficio. essendo una rete mesh, decide il centro stella chi debba collegarsi, dinamicamente, con chi. variabile nel tempo, chiaramente.)

**Reply by agnus777** [reply]
sono ripartito da una installazione di Z2M pulita ha fatto il paring di una decina di devices/router, ed ad un certo punto ha smesso di farlo se non dal coordinator. pensavo fosse il repeater di ikea, l'ho tolto ma non è cambiato niente. fatti riavvii vari, ma niente. 
Una volta fatto il paring dal coordinator, tutto va come deve andare.
Attenderò il prossimo aggiornamento di Z2M per vedere se è quello il problema

**Reply by KC-inDomus** [reply]
come sopra. non ha valore farlo, a prescindere, da router.

**Reply by agnus777** [reply]
Scusa ma non ho capito la tua risposta o forse non ho spiegato bene io il problema.

Ho una casa a 2 piani più un garage, il minipc con HA e l'antenna coordinator sono al piano primo. tramite una serie di devices router porto la rete mesh in tutte le zone di casa e all'esterno nel garage e piazzale.
Allo stato attuale, se provo a fare il pairing con un nuovo device dal garage o dal piano terra, non funziona. devo per forza andare al piano primo vicino al coordinator e fare il pairing che riesce senza problemi.
Una volta effettuato il pairing e portato ad esempio in garage, questo si aggancia alla rete mesh e tutto funziona scegliendosi i suoi punti di connessione.

che significa non ha valore farlo? più che altro come mai ha funzionato per i primi 10-11 devices e poi ha smesso?

**Reply by snakuzzo** [reply]
bella domanda...però devi farla a koenkk sul progetto di zigbee2mqtt, non qui

**Comment by trebestie** [comment]
Se cerchi
zigbee2mqtt force devices connect router
trovi parecchie discussioni, per esempio
https://community.home-assistant.io/t/zigbee2mqtt-force-devices-to-connect-to-another-router/387060/6
https://www.reddit.com/r/homeassistant/comments/t0tf14/tip_for_forcing_a_zigbee_device_to_connect_via_a/


Il giorno lun 29 apr 2024 alle ore 12:19 KC ***@***.***> ha
scritto:

**Reply by Scialla** [reply]
Nessuno ha detto che non si possa forzare ma, invece, che non ha senso farlo... questo perché la rete mesh zigbee gestisce autonomamente le connessioni. Questo vuol dire che se anche forzi la connessione con un router poi puoi trovarti il device collegato al coordinator o ad altro router, fra 1 minuto, 1 ora, 1 giorno, un mese etc.... la rete si gestisce da se
Altra cosa è invece che sia assodato esistano delle  incompatibilità tra marche differenti, o quantomeno delle difficoltà, a fare il pairing tra loro.
Meglio lasciar fare alla rete che si gestisce in autonomia le connessioni piuttosto che stare a litigare con i pairing diretti per poi vedere che fa quello che vuole

**Reply by snakuzzo** [reply]
@trebestie per cortesia non fare reply direttamente dalla mail.
Grazie
